### ISSUE TITLE
SCSSLint::Linter::BorderZero raised unexpected error

### ISSUE THREAD

**Issue by yoanmalie** [brigade/scss-lint#848]
Hi :wave: 
Here the issue:

```
SublimeLinter: scss: _carousel.scss ['/usr/bin/ruby', '-S', 'scss-lint', '--config', '/home/yoan/Projects/ge-cvm.fr/web/.scss-lint.yml'] 
SublimeLinter: scss output:
SCSSLint::Linter::BorderZero raised unexpected error linting file /tmp/SublimeLinter3-yoan/_carousel.scss: 'undefined method `[]' for nil:NilClass'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter/border_zero.rb:33:in `check_border'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter/border_zero.rb:26:in `visit_prop'
/var/lib/gems/2.3.0/gems/sass-3.4.21/lib/sass/tree/visitors/base.rb:36:in `visit'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter.rb:139:in `visit'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter.rb:150:in `block in visit_children'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter.rb:148:in `each'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter.rb:148:in `visit_children'
/var/lib/gems/2.3.0/gems/sass-3.4.21/lib/sass/tree/visitors/base.rb:38:in `visit'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter.rb:139:in `visit'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter.rb:150:in `block in visit_children'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter.rb:148:in `each'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter.rb:148:in `visit_children'
/var/lib/gems/2.3.0/gems/sass-3.4.21/lib/sass/tree/visitors/base.rb:36:in `block in visit'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter/border_zero.rb:21:in `visit_root'
/var/lib/gems/2.3.0/gems/sass-3.4.21/lib/sass/tree/visitors/base.rb:36:in `visit'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter.rb:139:in `visit'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/linter.rb:41:in `run'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/runner.rb:53:in `run_linter'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/runner.rb:35:in `block in find_lints'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/runner.rb:33:in `each'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/runner.rb:33:in `find_lints'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/runner.rb:19:in `block in run'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/runner.rb:18:in `each'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/runner.rb:18:in `run'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/cli.rb:67:in `scan_for_lints'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/cli.rb:59:in `act_on_options'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/lib/scss_lint/cli.rb:33:in `run'
/var/lib/gems/2.3.0/gems/scss_lint-0.50.2/bin/scss-lint:7:in `<top (required)>'
/usr/local/bin/scss-lint:23:in `load'
/usr/local/bin/scss-lint:23:in `<main>'
Report this bug at https://github.com/brigade/scss-lint/issues

To help fix this issue, please include:
- The above stack trace
- SCSS-Lint version: 0.50.2
- Sass version: 3.4.21
- Ruby version: 2.3.1
- Contents of /home/yoan/Projects/ge-cvm.fr/web/.scss-lint.yml 
```

The content of `/home/yoan/Projects/ge-cvm.fr/web/.scss-lint.yml` :
```
# Documentation:
# https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md

scss_files: 'assets/src/css/**/*.scss'
exclude:
  - node_modules
  - assets/src/css/vendors/**
linters:
  BangFormat:
    enabled: true
    space_before_bang: true
    space_after_bang: false
  BemDepth:
    enabled: true
    max_elements: 1
  BorderZero:
    enabled: true
    convention: 0
  ColorKeyword:
    enabled: true
  ColorVariable:
    enabled: false
  Comment:
    enabled: true
  DebugStatement:
    enabled: true
  DeclarationOrder:
    enabled: true
  DisableLinterReason:
    enabled: true
  DuplicateProperty:
    enabled: true
    ignore_consecutive:
        - background
        - transition
  ElsePlacement:
    enabled: true
    style: same_line
  EmptyLineBetweenBlocks:
    enabled: true
    ignore_single_line_blocks: true
  EmptyRule:
    enabled: true
  ExtendDirective:
    enabled: false
  FinalNewline:
    enabled: true
    present: true
  HexLength:
    enabled: true
    style: short
  HexNotation:
    enabled: true
    style: lowercase
  HexValidation:
    enabled: true
  IdSelector:
    enabled: true
  ImportantRule:
    enabled: false
  ImportPath:
    enabled: true
    leading_underscore: false
    filename_extension: false
  Indentation:
    enabled: true
    allow_non_nested_indentation: true
    character: space
    width: 4
  LeadingZero:
    enabled: true
    style: include_zero
  LengthVariable:
    enabled: false
  MergeableSelector:
    enabled: true
    force_nesting: false
  NameFormat:
    enabled: true
    allow_leading_underscore: true # For private variables, mixins and functions
    convention: '^[a-z]|[a-z0-9\-]{2,}$'
    convention_explanation: 'Valid names for variables, mixins and functions: i, foo, foo-bar, foo-1, _private-thing'
  NestingDepth:
    enabled: true
    max_depth: 3
    ignore_parent_selectors: false
  PlaceholderInExtend:
    enabled: false
  PrivateNamingConvention:
    enable: true
    prefix: '_'
  PropertyCount:
    enabled: false
  PropertySortOrder:
    enabled: false
  PropertySpelling:
    enabled: true
  PropertyUnits:
    enabled: true
    global: ['px', 'em', 'rem', '%', 'vh', 'vw', 'vmin', 'vmax', 'vm']
    properties:
      line-height: [] # No units allowed
  PseudoElement:
    enabled: true
  QualifyingElement:
    enabled: false
  SelectorDepth:
    enabled: true
    max_depth: 3
  SelectorFormat: # Note: this does not warn about malformed selector that start with #{$css-prefix}
    enabled: true
    convention: '^[a-z0-9_\-]+$' # a sort of strict_BEM
    convention_explanation: 'Valid class names: block_container, block, block--modifier, block__element, block__element--modifier'
  Shorthand:
    enabled: true
  SingleLinePerProperty:
    enabled: true
    allow_single_line_rule_sets: false
  SingleLinePerSelector:
    enabled: true
  SpaceAfterComma:
    enabled: true
    style: one_space
  SpaceAfterComment:
    enabled: true
    style: one_space
    allow_empty_comments: true
  SpaceAfterPropertyColon:
    enabled: true
    style: one_space
  SpaceAfterPropertyName:
    enabled: false
  SpaceAfterVariableColon:
    enabled: true
    style: at_least_one_space
  SpaceAfterVariableName:
    enabled: false
  SpaceAroundOperator:
    enabled: true
    style: one_space
  SpaceBeforeBrace:
    enabled: true
    allow_single_line_padding: true
    style: space
  SpaceBetweenParens:
    enabled: true
    spaces: 0
  StringQuotes:
    enabled: true
    style: single_quotes
  TrailingSemicolon:
    enabled: true
  TrailingWhitespace:
    enabled: true
  TrailingZero:
    enabled: true
  TransitionAll:
    enable: true
  UnnecessaryMantissa:
    enabled: true
  UnnecessaryParentReference:
    enabled: true
  UrlFormat:
    enabled: true
  UrlQuotes:
    enabled: true
  VariableForProperty:
    enabled: false
  VendorPrefix:
    enabled: true
  ZeroUnit:
    enabled: true

```

And the `_carousel.scss` file (not from `tmp/` but in my project) :
```
.slider {
    position: relative;
}

.slider__frame {
    position: relative;
    font-size: 0;
    line-height: 0;
    overflow: hidden;
    white-space: nowrap;
    background-color: palette('lightGrey');
    min-height: 20rem;
}

.frame-container {
    display: inline-block;
}

.frame__item {
    position: relative;
    display: inline-block;
}

// Buttons prev/next

.slider__prev,
.slider__next {
    display: block;
    position: absolute;
    top: 50%;
    height: 100%;
    width: 7rem;
    padding: 0 1rem;
    transform: translateY(-50%);
    cursor: pointer;
    border: none;
    outline: none;
    background: none;
}

.slider__prev {
    left: 0;
    text-align: left;

    &:hover svg + svg {
        fill: rgba(palette('lightGrey'), 0.5);
        right: 0.8rem;
        transition: right 200ms ease-in,
                    fill 200ms ease-in;
    }
}

.slider__next {
    right: 0;
    text-align: right;

    &:hover svg + svg {
        fill: rgba(palette('lightGrey'), 0.5);
        left: 0.8rem;
        transition: left 200ms ease-in,
                    fill 200ms ease-in;
    }
}

.slider__prev svg,
.slider__next svg {
    width: 3.5rem;
    height: 6rem;
    fill: rgba(palette('brand', 'lightGrey'), 0.5);

    + svg {
        position: absolute;
    }
}

.slider__prev svg + svg {
    right: 0;
    transition: right 100ms ease-out,
                fill 200ms ease-out;
}

.slider__next svg + svg {
    left: 0;
    transition: left, 100ms ease-out,
                fill 200ms ease-out;
}

```

Look like it's the line 36 !

**Comment by yoanmalie** [comment]
Hi,
today, i have this error:
`Fatal undefined`

when i remove the `border: none` on line 36, it's work :confused:
